### ISSUE TITLE
Go target, unable to check when custom error strategy is in recovery mode

### ISSUE THREAD

**Issue by seth2810** [antlr/antlr4#3259]
Hello, community!

I am trying to move existing parser error strategy from c++ to golang target:

```cpp
class MyErrorStrategy: public antlr4::DefaultErrorStrategy {
  void reportUnwantedToken(antlr4::Parser* recognizer) override {
    if (inErrorRecoveryMode(recognizer)) {
      return;
    }

    antlr4::Token* t = recognizer->getCurrentToken();
    std::string tokenName = getTokenErrorDisplay(t);
    antlr4::misc::IntervalSet expecting = getExpectedTokens(recognizer);

    throw CustomException({
      {"extraneous input " + tokenName, Position {int(t->getLine()), int(t->getCharPositionInLine()) + 1}},
      {" expecting " + expecting.toString(recognizer->getVocabulary()), Position()}
    });
  }
};
```
But I have a "small problem". Unlike c++, golang does not provide the ability to call private / unexported methods. This small difference makes it impossible to check when the strategy is `inErrorRecoveryMode`.

I saw that the strategy method `GetTokenErrorDisplay` was [made](https://github.com/antlr/antlr4/blob/b358b509191a0e076193cee6685256f9df0c7fc0/runtime/Go/antlr/error_strategy.go#L550-L563) public, [as well](https://github.com/antlr/antlr4/blob/b358b509191a0e076193cee6685256f9df0c7fc0/runtime/Go/antlr/parser.go#L406-L408) as the `parser` method `GetCurrentToken` [in contrast](https://github.com/antlr/antlr4/blob/e404d26f614291c0dbace9f42bf77cd435919271/runtime/Cpp/runtime/src/DefaultErrorStrategy.cpp#L271-L284) to the c++ runtime, where it has private access. Perhaps it's time to also make the `inErrorRecoveryMode` method public? I find an explanation why the methods of controlling the state flag are private, but I don't understand why method for reading it has private access.

I also assume that there is another way to do this check. If I'm right, please tell me how I can do it.

**Comment by seth2810** [comment]
@pboyer FYI

**Comment by seth2810** [comment]
@ericvergnaud

**Comment by xiaozuo7** [comment]
I had the same problem, which left me wondering how to catch errors @parrt

**Comment by parrt** [comment]
That sounds like the method should be public.  @jcking do you have any objection to making `inErrorRecoveryMode` public?

**Comment by jcking** [comment]
Yeah, it looks like it should be compared to other runtimes. SGTM

**Comment by parrt** [comment]
Weird. that it did not auto close to spite me saying "Fixes #3259". hmm...
